src/Workspaces/CSharp/Portable/Classification/ClassificationHelpers.cs

```diff
@@ -118,6 +119,48 @@ private static string GetClassificationForIdentifier(SyntaxToken token)
             {
                 return ClassificationTypeNames.TypeParameterName;
             }
+            else if (token.Parent is MethodDeclarationSyntax methodDeclaration && methodDeclaration.Identifier == token)
+            {
+                // TODO: Add proper extension method check - I'm sure there is some syntax based check somewhere burried in roslyn but I couldn't find it.
+                bool isExtensionMethod =
+                    (methodDeclaration.Parent as TypeDeclarationSyntax)?.Modifiers.Any(m => m.IsKind(SyntaxKind.StaticKeyword)) == true &&
```

Comment:
SyntaxTokenList already has a .Any overload that just takes a SyntaxKind.  So no need to use the lambda version.

Reply:
All you have to do is check the modifiers of the first parameter. That's sufficient.  "this" is only legal on extension methods, so it's fine to assume that if you see 'this' on the first parameter, then ti's an extension method.

Reply:
Is there any way checking that ? there must be something to check that but I didn't find it.